### PR TITLE
[Merged by Bors] - feat: make isNoetherian_of_isNoetherianRing_of_finite an instance

### PR DESCRIPTION
**Pull request by riccardobrasca**
Make `isNoetherian_of_isNoetherianRing_of_finite an instance`: this was impossible in Lean 3 because of a loop.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Probability/Cdf.lean [hunk 1]
```diff
@@ -69,8 +69,8 @@ lemma tendsto_cdf_atBot : Tendsto (cdf μ) atBot (𝓝 0) := tendsto_condCdf_atB
 /-- The cdf tends to 1 at +∞. -/
 lemma tendsto_cdf_atTop : Tendsto (cdf μ) atTop (𝓝 1) := tendsto_condCdf_atTop _ _
 
-set_option synthInstance.maxHeartbeats 25000 in
 lemma ofReal_cdf [IsProbabilityMeasure μ] (x : ℝ) : ENNReal.ofReal (cdf μ x) = μ (Iic x) := by
+  have := IsProbabilityMeasure.toIsFiniteMeasure (Measure.prod (Measure.dirac ()) μ)
```
Comment (riccardobrasca) [inline Mathlib/Probability/Cdf.lean:73]: `set_option synthInstance.maxHeartbeats 50000 in` also works here, but `25000` is not enough anymore.

**Comment by eric-wieser** [comment]
!bench

File: Mathlib/Probability/Cdf.lean (hunk 1 above)
Comment (ericrbg) [reply Mathlib/Probability/Cdf.lean:73]: really crazy that it affects something here.

**Comment by ericrbg** [comment]
can we try remove all explicit calls of `isNoetherian_of_isNoetherianRing_of_finite` to see if it actually works in practice?

**Comment by riccardobrasca** [comment]
> can we try remove all explicit calls of `isNoetherian_of_isNoetherianRing_of_finite` to see if it actually works in practice?

Let me see.

**Comment by riccardobrasca** [comment]
!bench

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**